common/app/navigation/helpers/UrlHelpers.scala

```diff
@@ -1,9 +1,91 @@
 package navigation
 
-import conf.Configuration.id.membershipUrl
+import conf.Configuration
+import play.api.libs.json.Json
 import play.api.mvc.RequestHeader
+import common.Edition
 
 object UrlHelpers {
+  sealed trait ReaderRevenueSite {
+    val url: String
+  }
+  case object Membership extends ReaderRevenueSite {
+    val url: String = s"${Configuration.id.membershipUrl}/supporter"
+  }
+  case object Contribute extends ReaderRevenueSite {
+    val url: String = Configuration.id.contributeUrl
+  }
+  case object Subscribe extends ReaderRevenueSite {
+    val url: String = Configuration.id.subscribeUrl
+  }
+
+  sealed trait Position
+  case object NewHeader extends Position
+  case object OldHeader extends Position
+  case object SideMenu extends Position
+  case object AmpHeader extends Position
+  case object Footer extends Position
+
+  def getCampaignCode(implicit request: RequestHeader, destination: ReaderRevenueSite, position: Position): String = {
```

Comment:
why is `destination` and `position` implicit arguments?

Reply:
ah, it should be just `request` - I forgot that `implicit` marks the whole param list as implicit